### PR TITLE
Cap project tools to netcoreapp2.2 and block creating new package

### PR DESCRIPTION
**Pull request by wli3**
2 work are separate commits

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets [hunk 1]
```diff
@@ -38,8 +38,8 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(DotnetCliToolTargetFramework)' == '' And '$(BundledNETCoreAppTargetFrameworkVersion)' != ''">
-    <!-- Set the TFM used to restore .NET CLI tools to match the version of .NET Core bundled in the CLI -->
-    <DotnetCliToolTargetFramework>netcoreapp$(BundledNETCoreAppTargetFrameworkVersion)</DotnetCliToolTargetFramework>
+    <!-- To move away from project tools. The highest available project tools TFM is netcoreapp2.2 -->
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets]: The full spec is not available to public yet

**Comment by wli3** [comment]
pending remove --fx-version when run, but that change should be in CLI repo

File: src/Tasks/Common/Resources/Strings.resx [hunk 2]
```diff
@@ -489,4 +489,8 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1092: The CLSIDMap cannot be embedded on the COM host because adding resources requires that the build be performed on Windows (excluding Nano Server).</value>
     <comment>{StrBegin="NETSDK1092: "}</comment>
   </data>
+  <data name="ProjectToolOnlySupportTFMLowerThanNetcoreapp22" xml:space="preserve">
+    <value>NETSDK1093: Project tools(DotnetCliTool) only support Target Framework Moniker lower than or equal to netcoreapp2.2.</value>
```
Comment (dsplaisted) [inline src/Tasks/Common/Resources/Strings.resx]: Add a space here: "tools (DotnetCliTool)"

Comment (dsplaisted) [reply src/Tasks/Common/Resources/Strings.resx]: I'd reword this to:

> NETSDK1093: Project tools (DotnetCliTool) only support targeting .NET Core 2.2 and lower.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets]: I would say this:

> Maximum supported target framework for DotnetCliProjectTools is .NET Core 2.2

File: src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToPackAProjectTool : SdkTest
+    {
+        public GivenThatWeWantToPackAProjectTool(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Fact]
+        public void It_packs_project_tools_has_tfm_lower_or_equal_to_netcoreapp2_2()
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs]: I'd name this `It_packs_project_tools_targeting_netcoreapp2_2`

File: src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            };
+
+            toolProject.AdditionalProperties.Add("PackageType", "DotnetCliTool");
+
+            var asset = _testAssetsManager
+                       .CreateTestProject(toolProject, toolProject.Name)
+                       .Restore(Log, toolProject.Name);
+
+            var result = new PackCommand(Log, Path.Combine(asset.TestRoot, toolProject.Name)).Execute();
+            result.ExitCode.Should().NotBe(0);
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs]: You can write these two checks as something like this:

```c#
result
    .Should()
    .Fail()
    .And.HaveStdOutContaining(Strings.ProjectToolOnlySupportTFMLowerThanNetcoreapp22);
```

File: src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            var asset = _testAssetsManager
+                       .CreateTestProject(toolProject, toolProject.Name)
+                       .Restore(Log, toolProject.Name);
+
+            var packCommand = new PackCommand(Log, Path.Combine(asset.TestRoot, toolProject.Name));
+            packCommand.Execute().Should().Pass();
+        }
+
+        [Fact]
+        public void It_packs_project_tools_has_tfm_higher_than_netcoreapp2_2()
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackAProjectTool.cs]: Rename this to `It_fails_to_pack_project_tools_targeting_netcoreapp3_0`

File: src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreAndGenerateDepsjsonDotNetCliToolReference.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                                              FolderNameForRestorePackages,
+                                              ".tools",
+                                              CodeGenerationPackageName,
+                                              ProjectToolVersion,
+                                              ExpectedProjectToolRestoreTargetFrameworkMoniker,
+                                              "project.assets.json");
+            LockFile lockFile = LockFileUtilities.GetLockFile(assetsJsonPath, NullLogger.Instance);
+            lockFile.Targets.Single().TargetFramework
+                .Should().Be(NuGetFramework.Parse(ExpectedProjectToolRestoreTargetFrameworkMoniker),
+                "Restore target framework should be caped at netcoreapp2.2 due to moving away from project tools." +
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreAndGenerateDepsjsonDotNetCliToolReference.cs]: Spelling: capped

**Review by dsplaisted — APPROVED** [review]

File: src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreDotNetCliToolReference.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        private const string ProjectToolVersion = "1.0.0";
+        private const string ExpectedProjectToolRestoreTargetFrameworkMoniker = "netcoreapp2.2";
+
+        public GivenThatWeWantToRestoreDotNetCliToolReference(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Fact]
+        public void It_can_restore_with_netcoreapp2_2()
```
Comment (wli3) [inline src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreDotNetCliToolReference.cs]: @dsplaisted I find previous test restoring aspnet scaffolding is too slow and download too much, and may cause fullframework leg time out. So I changed it to restore a sample project tool and only asset restore's TFM
